client/src/components/accordion/component.tsx

```diff
@@ -0,0 +1,81 @@
+import { ChevronRightIcon } from '@heroicons/react/solid';
+import classNames from 'classnames';
+import type { HTMLAttributes } from 'react';
+import { useCallback, useEffect, useState, useMemo } from 'react';
+
+type AccordionEntryProps = React.PropsWithChildren<{
+  header: React.ReactNode;
+  expanded?: boolean;
+  onExpandedChange?: (expanded: boolean) => void;
+}>;
+
+type AccordionProps = Omit<HTMLAttributes<HTMLDivElement>, 'children'> &
```

Comment:
you have https://headlessui.com/react/disclosure available in HeadlessUi, did you check it? in that case, why did you decide to make a custom one?

Reply:
I didn't know headless UI had one, I can probably quickly swap it